sdks/java/container/boot.go

```diff
@@ -153,24 +153,19 @@ func main() {
 			cp = append(cp, filepath.Join(jarsDir, jar))
 		}
 	}
-	cp = append(cp, filepath.Join(jarsDir, javaHarnessJar))
-
 	var hasWorkerExperiment = strings.Contains(options, "use_staged_dataflow_worker_jar")
+
+	if !hasWorkerExperiment {
+		cp = append(cp, filepath.Join(jarsDir, javaHarnessJar))
+	} else {
+		// Skip adding system "beam-sdks-java-harness.jar". User-provided jar will
```

Comment:
for my education, is this internal-only flow?

Reply:
external user could also use it. The purpose is detailed in the description of #33508users without docker access could test/run custom build beam-sdk-harness now